taskvine/src/bindings/python3/ndcctools/taskvine/manager.py

```diff
@@ -949,16 +970,18 @@ def create_library_from_functions(self, name, *function_list, poncho_env=None, i
                 need_pack = False
 
             # create library code and environment, if appropriate
-            package_serverize.serverize_library_from_code(library_cache_path, function_list, name, need_pack=need_pack, import_modules=import_modules)
+            package_serverize.serverize_library_from_code(library_cache_path, function_list, library_name, need_pack=need_pack, import_modules=import_modules)
 
             # enable correct permissions for library code
             os.chmod(library_code_path, 0o775)
 
         # Create Task to execute the Library and prepend it with some setup code if needed.
         if init_command:
-            t = LibraryTask(f"{init_command} python ./library_code.py", name)
+            t = LibraryTask(f"{init_command} python ./library_code.py", library_name)
         else:
-            t = LibraryTask("python ./library_code.py", name)
+            t = LibraryTask("python ./library_code.py", library_name)
+        function_names = [function.__name__ for function in function_list]
```

Comment:
Functions may be passed already as strings.  Also, function names should probably be part of the `LibraryTask` constructor, as it is fundamental for its purpose (`LibraryTask(cmd, *functions)`). In this way you do not need the function to set library names, which if called after the construction of the library will create inconsistencies.

Reply:
I now pass new arguments `function_list` as well as `library_code_path` for the sake of consistency. Thank you!